### ISSUE TITLE
Port libdrm

### ISSUE THREAD

**Issue by X547** [haikuports/haikuports#4745]
Libdrm is required by Mesa for hardware acceleration support.

Repository: https://gitlab.freedesktop.org/mesa/drm.

**Comment by waddlesplash** [comment]
It's just a lot of ioctl hooks, and IIRC when I asked Mesa devs about it before, they said that a lot of drivers just call the ioctls directly. Plus it is of course useless without kernel DRM drivers.

If you intend to work on graphics acceleration "for real", please do ping me on IRC to coordinate our efforts, especially if you need help with the kernel side of things :)

**Comment by X547** [comment]
At first I want to identify required dependencies and build mesa userland part even it will not actually work. Driver code contains many Linux-specific extensions that need to be fixed for successful build.

**Comment by waddlesplash** [comment]
Yes, I have a strategy for dealing with all of that. I will dive into this again over the next few days; just find me on IRC whenever you are ready to coordinate efforts.
